### ISSUE TITLE
using wire as protoc plugin in a protorepo

### ISSUE THREAD

**Issue by mgenov** [square/wire#1804]
Is it possible wire to be used as protoc plugin in a protorepo? 

The overall goal is proto files to be compiled using some kind of wire_out arguments to protoc?

Such feature will make wire easily integrated with tools such as: https://github.com/grpckit/omniproto

Similar question:
https://stackoverflow.com/questions/64103974/using-wire-in-protorepo

**Comment by swankjesse** [comment]
In our environment at Square we use Wire + Gradle to build JVM targets, and protoc to build other platforms. It’s more configuration, but not that much more.

**Comment by JakeWharton** [comment]
Also Swift targets now! 😁

**Comment by jan-xyz** [comment]
@swankjesse is there a way that one could contribute a protoc plugin?

**Comment by swankjesse** [comment]
@jan-xyz how would it work? (I don’t want this project to include C++ code!)

**Comment by jan-xyz** [comment]
TBH, I haven't looked into it yet. I know that the Go extension is written in Go, so I'd assume that the plug-in could also be written in Java or Kotlin. I didn't want to spend too much time of your answer was a clear 'no'. As it stands right now the wire plug-in takes ~15 minutes to build our protobuf artifacts while other languages are done within ~2-5 minutes and I was hoping for a performance increase similar to the ScalaPB increase when switching from the standalone shipped compiler to the protoc plugin.

**Comment by swankjesse** [comment]
15 minutes?!!! Wow!

Wire’s compiler is simple & single threaded. We recently looked at introducing concurrency but figured that the total amount of time saved wouldn’t justify the increased complexity.

If you've got 15-minute builds then there’s a better fix than switching compilers; we can just do the work make Wire use resources better.

Internal to Square we've got a giant corporation-sized protos directory that we currently build as independent modules. We could try to build this in one huge step and see how long that takes, and perhaps use that to approximate your situation.
